### PR TITLE
Add xml comments for Language.AST namespace

### PR DESCRIPTION
**Pull request by Shane32**
I plowed through this as fast as I could possibly go.  Hopefully it's accurate...enough.

### CODE REVIEW

File: src/GraphQL/Language/AST/INode.cs [hunk 1]
```diff
@@ -2,12 +2,25 @@
 
 namespace GraphQL.Language.AST
 {
+    /// <summary>
+    /// Represents a node within a document.
+    /// </summary>
     public interface INode
     {
+        /// <summary>
+        /// Returns a list of children nodes.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/INode.cs:11]:
```suggestion
        /// Returns a list of children nodes. If the node doesn't have children, returns <see langword="null"/>.
```

File: src/GraphQL/Language/AST/CommentNode.cs [hunk 2]
```diff
@@ -2,17 +2,30 @@
...
             Value = value;
         }
 
+        /// <summary>
+        /// Returns the comment stored in this node.
+        /// </summary>
         public string Value { get; }
 
+        /// <summary>
+        /// Compares this instance to another <see cref="CommentNode"/> by name.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/CommentNode.cs:24]:
```suggestion
        /// Compares this instance to another <see cref="CommentNode"/> by comment value.
```

**Comment by Shane32** [comment]
> If you can review this tonight @sungam3r then I can spend some time tonight on a new PR targeting develop to clean up some issues I found while adding these comments.

This is the one I want to work on tonight!!

**Comment by sungam3r** [comment]
Perhaps I will be in time, I have one more hour.

File: src/GraphQL/Language/AST/FragmentDefinition.cs [hunk 3]
```diff
@@ -3,23 +3,44 @@
...
 {
+    /// <summary>
+    /// Represents a fragment definition node within a document.
+    /// </summary>
     public class FragmentDefinition : AbstractNode, IDefinition, IHaveSelectionSet
     {
+        /// <summary>
+        /// Initializes a new fragment definition node with the specified <see cref="NameNode"/> containing the name of this fragment definition.
+        /// </summary>
+        /// <param name="node"></param>
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/FragmentDefinition.cs:14]: ?

File: src/GraphQL/Language/AST/ListType.cs [hunk 4]
```diff
@@ -2,23 +2,35 @@
...
 {
+    /// <summary>
+    /// Represents a list type node within a document.
+    /// </summary>
     public class ListType : AbstractNode, IType
     {
+        /// <summary>
+        /// Initializes a list type node that wraps the specified type node.
+        /// </summary>
+        /// <param name="type"></param>
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/ListType.cs:13]: ?

File: src/GraphQL/Language/AST/NullValue.cs [hunk 5]
```diff
@@ -1,12 +1,16 @@
 namespace GraphQL.Language.AST
 {
+    /// <summary>
+    /// Represents the value null within a document.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/NullValue.cs:4]:
```suggestion
    /// Represents the 'null' value within a document.
```

File: src/GraphQL/Language/AST/OperationType.cs [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
     {
+        /// <summary>
+        /// A query operation.
+        /// </summary>
         Query,
+        /// <summary>
+        /// A mutation operation.
+        /// </summary>
         Mutation,
+        /// <summary>
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/OperationType.cs:16]: Let's separate enumeration values by new line.

File: src/GraphQL/Language/AST/SelectionSet.cs [hunk 7]
```diff
@@ -17,32 +23,46 @@ private SelectionSet(List<ISelection> selections)
...
+        /// <summary>
+        /// Adds a node to the start of the list.
+        /// </summary>
         public void Prepend(ISelection selection)
         {
             SelectionsList.Insert(0, selection ?? throw new ArgumentNullException(nameof(selection)));
         }
 
+        /// <summary>
+        /// Adds a node to the list.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/SelectionSet.cs:46]:
```suggestion
        /// Adds a node to the end of the list.
```

File: src/GraphQL/Language/AST/SourceLocation.cs [hunk 8]
```diff
@@ -10,22 +16,38 @@ public SourceLocation(int line, int column, int start = -1, int end = -1)
...
+        /// Returns the column number within the query text.
+        /// </summary>
         public int Column { get; }
 
-        /// <inheritdoc />
+        /// <inheritdoc/>
         public override string ToString() => $"line={Line}, column={Column}, start={Start}, end={End}";
 
+        /// <summary>
+        /// Compares this instance to another based on the line and column values.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/SourceLocation.cs:43]: Why only line and column ?? I don't know.

File: src/GraphQL/Language/AST/SourceLocation.cs [hunk 9]
```diff
@@ -37,12 +59,7 @@ public override bool Equals(object obj)
...
 
-        public override int GetHashCode()
-        {
-            unchecked
-            {
-                return (Line * 397) ^ Column;
-            }
-        }
+        /// <inheritdoc/>
+        public override int GetHashCode() => unchecked((Line * 397) ^ Column);
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/SourceLocation.cs:63]:
```suggestion
        public override int GetHashCode() => (Line, Column).GetHashCode();
```

File: src/GraphQL/Language/AST/ValueNodes/EnumValue.cs [hunk 10]
```diff
@@ -2,30 +2,49 @@
...
+        /// Returns a <see cref="NameNode"/> containing the string representation of the enumeration value.
+        /// </summary>
         public NameNode NameNode { get; }
 
-        /// <inheritdoc />
+        /// <inheritdoc/>
         public override string ToString() => $"EnumValue{{name={Name}}}";
 
+        /// <summary>
+        /// Compares this instances to another instance by comparing the string representation of the enumeration value.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/ValueNodes/EnumValue.cs:43]:
```suggestion
        /// Compares this instance to another instance by comparing the string representation of the enumeration value.
```

File: src/GraphQL/Language/AST/ValueNodes/ObjectField.cs [hunk 11]
```diff
@@ -3,34 +3,59 @@
 
 namespace GraphQL.Language.AST
 {
+    /// <summary>
+    /// Represents the child field node of a object value node within a document.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/ValueNodes/ObjectField.cs:7]:
```suggestion
    /// Represents the child field node of an object value node within a document.
```

File: src/GraphQL/Language/AST/ValueNodes/ObjectValue.cs [hunk 12]
```diff
@@ -14,6 +20,9 @@ public ObjectValue(IEnumerable<ObjectField> fields)
                 ObjectFields = fields;
         }
 
+        /// <summary>
+        /// Returns a <see cref="Dictionary{TKey, TValue}">Dictionary&lt;string, object&gt;</see> containing the values of the field nodes that this object value node contains.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/ValueNodes/ObjectValue.cs:24]:
```suggestion
        /// Returns a <see cref="Dictionary{TKey, TValue}">Dictionary&lt;string, object&gt;</see>
        /// containing the values of the field nodes that this object value node contains.
```

File: src/GraphQL/Language/AST/ValueNodes/ValueNode.cs [hunk 13]
```diff
@@ -1,14 +1,19 @@
 namespace GraphQL.Language.AST
 {
+    /// <summary>
+    /// Represents a value node which contains an literal value within a document.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/ValueNodes/ValueNode.cs:4]:
```suggestion
    /// Represents a value node which contains a literal value within a document.
```

File: src/GraphQL/Language/AST/VariableDefinition.cs [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+        /// </summary>
+        public NameNode NameNode { get; set; }
+
+        /// <summary>
+        /// Returns the type node representing the graph type of the variable.
+        /// </summary>
+        public IType Type { get; set; }
+
+        /// <summary>
+        /// Returns a value node representing the default value of the variable.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/VariableDefinition.cs:42]:
```suggestion
        /// Returns a value node representing the default value of the variable.
        /// Returns <see langword="null"/> if the variable has no default value. 
```

File: src/GraphQL/Language/AST/VariableReference.cs [hunk 15]
```diff
@@ -0,0 +1,36 @@
+using System;
+
+namespace GraphQL.Language.AST
+{
+    public class VariableReference : AbstractNode, IValue
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/VariableReference.cs:5]: Comments?

File: src/GraphQL/Language/AST/Variables.cs [hunk 16]
```diff
@@ -5,10 +5,16 @@
 
 namespace GraphQL.Language.AST
 {
+    /// <summary>
+    /// Contains a list of variables (name &amp; value tuples) that have been gathered from the document and attached <see cref="Inputs"/>.
```
Comment (sungam3r) [inline src/GraphQL/Language/AST/Variables.cs:9]: attached **to** ?

File: src/GraphQL/Language/AST/FragmentDefinition.cs (hunk 3 above)
Comment (Shane32) [reply src/GraphQL/Language/AST/FragmentDefinition.cs:14]:
```suggestion
```

File: src/GraphQL/Language/AST/ListType.cs (hunk 4 above)
Comment (Shane32) [reply src/GraphQL/Language/AST/ListType.cs:13]:
```suggestion
```

File: src/GraphQL/Language/AST/NamedType.cs [hunk 17]
```diff
@@ -2,25 +2,42 @@
...
 {
+    /// <summary>
+    /// Represents a named type node within a document.
+    /// </summary>
     public class NamedType : AbstractNode, IType
     {
+        /// <summary>
+        /// Initializes a new named type node containing the specified <see cref="NameNode"/>.
+        /// </summary>
+        /// <param name="node"></param>
```
Comment (Shane32) [inline src/GraphQL/Language/AST/NamedType.cs:13]:
```suggestion
```

File: src/GraphQL/Language/AST/OperationType.cs [hunk 18]
```diff
@@ -1,9 +1,21 @@
...
+    /// <summary>
+    /// An enumeration of the GraphQL operation types.
+    /// </summary>
     public enum OperationType
     {
+        /// <summary>
+        /// A query operation.
+        /// </summary>
         Query,
+        /// <summary>
```
Comment (Shane32) [inline src/GraphQL/Language/AST/OperationType.cs:12]:
```suggestion

        /// <summary>
```

File: src/GraphQL/Language/AST/OperationType.cs (hunk 6 above)
Comment (Shane32) [reply src/GraphQL/Language/AST/OperationType.cs:16]:
```suggestion

        /// <summary>
```

File: src/GraphQL/Language/AST/Variables.cs (hunk 16 above)
Comment (Shane32) [reply src/GraphQL/Language/AST/Variables.cs:9]: Not in this case.  So it's not "attaching a rope to a tree"  but rather more like "Please open the attached files."

File: src/GraphQL/Language/AST/VariableReference.cs (hunk 15 above)
Comment (Shane32) [reply src/GraphQL/Language/AST/VariableReference.cs:5]: whoops

Comment (Shane32) [reply src/GraphQL/Language/AST/VariableReference.cs:5]: I'll do it in a separate PR

**Comment by Shane32** [comment]
@sungam3r I'm merging this now.  Of course, let me know if you've notice any additional changes and I can make them in another PR.

File: src/GraphQL/Language/AST/Variables.cs (hunk 16 above)
Comment (sungam3r) [reply src/GraphQL/Language/AST/Variables.cs:9]: ok

**Comment by sungam3r** [comment]
I wrote about everything I found.

**Pull request merged**